### PR TITLE
Add entry via stdin to the init command

### PR DESCRIPTION
**Pull request by brunats**
**- What I did**
Add entry via stdin to the init command. - Also includes stdin tests for init - Updates tests from the init_test.go file to include stdin tests.

**- How to verify it**

Clone this pr with the command and build Ritchie
```bash
git fetch upstream pull/426/head:feature/add-entry-via-stdin-to-the-init-command
git checkout feature/add-entry-via-stdin-to-the-init-command
make build && sudo cp dist/linux/rit /usr/local/bin/
```

Run tests
 ```bash
make functional-test # Search for "Running STDIN: Init STDIN"
make unit-test # Search for "=== RUN   TestNewInitStdin"
```

**- Description for the changelog**

Add entry via stdin to the init command.

### CODE REVIEW

**Comment by lucasdittrichzup** [comment]
> **- What I did**
>
> **- How to verify it**
>
> **- Description for the changelog**

I think you forgot to put the explanatory texts in the PR

**Comment by brunasilvazup** [comment]
> > **- What I did**
> > **- How to verify it**
> > **- Description for the changelog**
> 
> I think you forgot to put the explanatory texts in the PR

Yes, I am still developing it, but I will remember to update the description.

Thanks (:

File: pkg/cmd/init.go [hunk 1]
```diff
@@ -122,6 +127,60 @@ func (in initCmd) runPrompt() CommandRunnerFunc {
 	}
 }
 
+func (in initCmd) runStdin() CommandRunnerFunc {
+	return func(cmd *cobra.Command, args []string) error {
+		init := initStdin{}
+
+		err := stdin.ReadJson(cmd.InOrStdin(), &init)
```
Comment (brunasilvazup) [inline pkg/cmd/init.go:134]: This `cmd.InOrStdin()` returns `os.Stdin()` by default, so it continues with the normal behavior for use in production and we were able to mock it for testing.

File: pkg/cmd/init_test.go [hunk 2]
```diff
@@ -38,24 +41,59 @@ func TestNewSingleInitCmd(t *testing.T) {
...
+		t.Errorf("NewInitCmd got %v", cmd)
+		return
+	}
+
+	if err := cmd.Execute(); err != nil {
+		t.Errorf("%s = %v, want %v", cmd.Use, err, nil)
+	}
+}
+
+func Test_initCmd_runAnyEntry(t *testing.T) {
```
Comment (brunasilvazup) [inline pkg/cmd/init_test.go:62]: I did stdin's tests with cmd's, so we can cover the situations.

File: pkg/cmd/init.go (hunk 1 above)
Comment (brunats) [inline pkg/cmd/init.go:134]: This cmd.InOrStdin() returns os.Stdin() by default, so it continues with the normal behavior for use in production and we were able to mock it for testing.

File: pkg/cmd/init_test.go (hunk 2 above)
Comment (brunats) [inline pkg/cmd/init_test.go:62]: I did stdin's tests with cmd's, so we can cover the situations.

File: pkg/cmd/init.go [hunk 3]
```diff
@@ -122,6 +127,60 @@ func (in initCmd) runPrompt() CommandRunnerFunc {
...
+		if err != nil {
+			return err
+		}
+
+		if init.AddRepo == "no" {
+			fmt.Println()
+			prompt.Warning(addRepoInfo)
+			fmt.Println()
```
Comment (victor-schumacher) [inline pkg/cmd/init.go:142]: The purpose here is to add lines between addRepoInfo?
What did you think about the following implementation?
``` 
fmt.Printf("\n%s\n", prompt.Yellow(addRepoInfo))
```

Comment (brunasilvazup) [reply pkg/cmd/init.go]: Yes, I thought the idea was good.

Fix in https://github.com/ZupIT/ritchie-cli/pull/426/commits/53b22542c7bd904b295482fcc8753ad9d30c8e59

**Pull request merged**